homeworks/week3/Duy-hw.py

```diff
@@ -0,0 +1,63 @@
+class Node:
+    def __init__(self, val):
+        self.val = val 
+        self.next = None
+class SinglyLinkedList:
+    def __init__(self):
+        self.head = Node(float('-inf')) #sentinel node
+    def add(self, index, e):
+        '''Add e to specific index. Do nothing if index is OOR or e is None'''
+        #Time Complexity: O(N), Space: O(1)
+        if not e: #e == None
+            return
+        prev = self.head 
+        cur = self.head.next
```

Comment:
Your approach is correct. However, you can make the code shorter by pointing `cur` to the element before the position you want to insert. Thus, getting rid of the `prev` variable.